React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm

```diff
@@ -258,6 +293,7 @@ - (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &
   self.viewController.modalPresentationStyle = presentationConfiguration(newProps);
 
   _shouldPresent = newProps.visible;
+  [self handleDetent: newProps.detent];
```

Comment:
From @sammy-SC 

> should we check if detent changes?

I think we should check whether the new detent is different from the old one and proceed updating the props only if the two differs.

What do you think?

Reply:
Hey so I tested by comparing old and new props but it turns out we can’t really add that guard. The reason is, whenever the modal closes, `prepareForRecycle` gets called and it sets `_viewController` to nil. With this we loses the detents we set on the `_viewController`. So when the next time we open the modal,
detents aren’t applied due to the guard we set previously and `_viewController` has already been reset and lost the detents on it. 

Here's how the guard is added:

<img width="879" alt="Screenshot 2022-10-18 at 12 38 12 PM" src="https://user-images.githubusercontent.com/47336142/196366780-1861dcd9-f5e7-46eb-a8ae-6b6f22011e17.png">

Reply:
Thank you for checking this!